### ISSUE TITLE
Invalid autocorrect between `Style/ArgumentsForwarding` and `Style/MethodDefParentheses`

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#15360]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/ArgumentsForwarding / Style/MethodDefParentheses:

### Steps to reproduce the problem

```
echo "def draw_point **opts; draw_rect(**opts); end" | RUBOCOP_TARGET_RUBY_VERSION=4.0 bundle exec rubocop -s test.rb --only Style/ArgumentsForwarding,Style/MethodDefParentheses -a

Inspecting 1 file
F

Offenses:

test.rb:1:16: C: [Corrected] Style/ArgumentsForwarding: Use anonymous keyword arguments forwarding (**).
def draw_point **opts; draw_rect(**opts); end
               ^^^^^^
test.rb:1:16: C: [Corrected] Style/MethodDefParentheses: Use def with parentheses when there are parameters.
def draw_point **opts; draw_rect(**opts); end
               ^^^^^^
test.rb:1:19: F: Lint/Syntax: unexpected token )
(Using Ruby 4.0 parser; configure using TargetRubyVersion parameter, under AllCops)
def draw_point(**)); draw_rect(**); end
                  ^
test.rb:1:34: C: [Corrected] Style/ArgumentsForwarding: Use anonymous keyword arguments forwarding (**).
def draw_point **opts; draw_rect(**opts); end
                                 ^^^^^^

1 file inspected, 4 offenses detected, 3 offenses corrected
====================
def draw_point(**)); draw_rect(**); end
```

### Expected behavior

Valid synax after autocorrect

### Actual behavior

See above, adds an extra brace.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.86.2 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 4.0.4) [x86_64-linux]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.2
```
